From 4b4fd0f6b0264b52e1f70d5a43790594acb06479 Mon Sep 17 00:00:00 2001
From: Alistair Strachan <alistair.strachan@imgtec.com>
Date: Wed, 11 Mar 2015 12:13:17 -0700
Subject: [PATCH 573/634] seccomp: really fix syscall numbers for x86 and
 x86_64

This fixes a build break on x86/x86_64 introduced by f2caf97
"seccomp: fix syscall numbers for x86 and x86_64".

The original change looks like it was cherry-picked from a kernel
with sys_renameat2, but since this kernel (android-3.14) lacks an
implementation of that syscall, it is not correct to refer to it
in the x86 syscall tables, or vmlinux will fail to link.

Do what ARM does and define the syscall slot but have it wired
up to the sys_ni_syscall / compat_ni_syscall so we retain syscall
binary compatibility with newer kernels.

Change-Id: I3fd1bc15557f0df3cf744bc8279b0985d48b93bf
Signed-off-by: Alistair Strachan <alistair.strachan@imgtec.com>
---
 arch/x86/syscalls/syscall_32.tbl | 2 +-
 arch/x86/syscalls/syscall_64.tbl | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/syscalls/syscall_32.tbl b/arch/x86/syscalls/syscall_32.tbl
index 7527eac..3934f1e 100644
--- a/arch/x86/syscalls/syscall_32.tbl
+++ b/arch/x86/syscalls/syscall_32.tbl
@@ -359,5 +359,5 @@
 350	i386	finit_module		sys_finit_module
 351	i386	sched_setattr		sys_sched_setattr
 352	i386	sched_getattr		sys_sched_getattr
-353	i386	renameat2		sys_renameat2
+353	i386	renameat2
 354	i386	seccomp			sys_seccomp
diff --git a/arch/x86/syscalls/syscall_64.tbl b/arch/x86/syscalls/syscall_64.tbl
index 16272a6..3ae4142 100644
--- a/arch/x86/syscalls/syscall_64.tbl
+++ b/arch/x86/syscalls/syscall_64.tbl
@@ -322,7 +322,7 @@
 313	common	finit_module		sys_finit_module
 314	common	sched_setattr		sys_sched_setattr
 315	common	sched_getattr		sys_sched_getattr
-316	common	renameat2		sys_renameat2
+316	common	renameat2
 317	common	seccomp			sys_seccomp
 
 #
-- 
1.9.1

